### PR TITLE
Yunchao/upcoming command

### LINKED ISSUE #73: feat: upcoming command
Upcoming comment would function similarly to `lastContacted` command whereby it would show on the GUI an updated list where it is sorted by people with upcoming fields.

## similarities
- people without upcoming fields would not be shown in filtered view

## difference
- people would be sorted by upcoming in ascending order of dates ie. closest appt would be shown the highest on the list

### PR DESCRIPTION
**Pull request by tanyunchao**
fixes #73 

## Details
- worked off @Colex2000 previous issue
- added `upcoming` command that sorts by ascending dates, people without upcoming are not shown
- known bug: default people have upcoming dates that have already passed which doesnt make sense, pending fix in #74 

image

### CODE REVIEW

File: src/main/java/seedu/address/model/person/LastContact.java [hunk 1]
```diff
@@ -49,17 +49,15 @@ public static boolean isValidDateTime(String dateTime) {
             return true;
         }
         String trimmedDateTime = dateTime.trim();
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-uuuu HHmm")
-                .withResolverStyle(ResolverStyle.STRICT);
         try {
-            LocalDateTime.parse(trimmedDateTime, formatter);
```
Comment (tanyunchao) [inline src/main/java/seedu/address/model/person/LastContact.java:55]: @Colex2000 please take note, i itchy finger and went to replace ur DateTimeFormatter here to the one you have already initiated as a CONST above

File: src/main/java/seedu/address/model/person/LastContact.java [hunk 2]
```diff
@@ -49,17 +49,15 @@ public static boolean isValidDateTime(String dateTime) {
...
+            LocalDateTime.parse(trimmedDateTime, DATETIME_FORMATTER);
             return true; // Successfully parsed, input matches the format
         } catch (DateTimeParseException e) {
             return false; // Parsing failed, input does not match the format
         }
     }
 
-    public boolean isLastContacted() {
```
Comment (tanyunchao) [inline src/main/java/seedu/address/model/person/LastContact.java:62]: @Colex2000 same here, i updated the function name to hasLastContacted to match the change in field name. changed all places that uses this method as well

File: src/main/java/seedu/address/model/person/LastContact.java (hunk 1 above)
Comment (Colex2000) [reply src/main/java/seedu/address/model/person/LastContact.java:55]: Thanks, mb I forgot to update.

File: src/main/java/seedu/address/model/person/LastContact.java (hunk 2 above)
Comment (Colex2000) [reply src/main/java/seedu/address/model/person/LastContact.java:62]: Can, thanks!

**Review by Colex2000 — COMMENTED** [review]
LGTM!

**Review by SimHanKiong — APPROVED** [review]
Commits very clean and easy to read. Good test coverage. Adding a known bug as an issue is good practice

**Pull request merged**